### PR TITLE
Blueprint integration

### PR DESCRIPTION
**Pull request by mkvanhooft**
Largest changes:

- Support for multi node simulation -> Mostly local changes regarding ports and choosing the correct EGP in netstack based on the request
- Blueprint integration, has multiple implications:
- 1) The building of the network is almost completely removed from SquidASM, only a conversion to some SquidASM objects is done
- 2) The network configuration is also not really in SquidASM anymore. The squidasm/run/config/py has some configs that mostly copy the netbuilder configs that allow for backward compatibility. I want to deprecate this in the future once I am more confident that the netbuilder configs are stable.
- 3) All classical communication go via a routing service from netsquid_driver. 
-  - For classical sockets this means it forwards to a netsquid classical socket object that has the same function, but the Netsquid socket is not a NetQASM socket. 
-  - For QNOS communication a QNOSNetworkService has been created that is mostly an interface that forwards messages to and from the routing service.
- Removed bell state correction in netstack. It seems that bell state correction is now built in into NetQASM itself.
- ProcessingNode has been renamed to StackNode as ProcessingNode is now a object in netsquid_netbuilder and the StackNode is an object in SquidASM that inherits from ProcessingNode, but adds the Host and QNOS components
- Examples and tests have been modified to use the netbuilder configs. Thus configs have been changed to use the words processingNode and node instead of stack (I need to think if this is a change that I want to make, since two sets of network configuration may be confusing for users)

### CODE REVIEW

File: .gitmodules
Comment (mkvanhooft) [inline .gitmodules:1]: Will be removed once netsquid-netbuilder is released

File: BETA-CHANGELOG.md
Comment (mkvanhooft) [inline BETA-CHANGELOG.md:1]: TODO: absorb into changelog

File: Makefile
Comment (mkvanhooft) [inline Makefile:1]: Will be removed once netsquid-netbuilder is released

File: pyproject.toml
Comment (mkvanhooft) [inline pyproject.toml:1]: Put this back

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,7 @@
 [metadata]
 name = squidasm
+version = 0.0.4
```
Comment (mkvanhooft) [inline setup.cfg]: Remove this

File: squidasm/sim/stack/netstack.py [hunk 2]
```diff
@@ -325,42 +318,20 @@ def handle_create_ck_request(
...
+                sender=current_egp, signal_label=ResCreateAndKeep.__name__
             )
             # Get the EGP's result.
-            result: ResCreateAndKeep = self._egp.get_signal_result(
+            result: ResCreateAndKeep = current_egp.get_signal_result(
                 ResCreateAndKeep.__name__, receiver=self
             )
             self._logger.info(f"got result for pair {pair_index}: {result}")
 
-            # This code is commented out for a hotfix as it was found that heralded link did not return
```
Comment (bvdvecht) [inline squidasm/sim/stack/netstack.py:341]: I see this code is removed completely now. Is this something that is planned to be put back in?

**Review by bvdvecht — COMMENTED** [review]
Overall, looks good! I so no real issues with any of the changes. I put in one small question about the bell state corrections.

Besides that, I still see a bunch of TODOs related to the actual release (like version numbers, using git submodule, the beta changelog etc). I'd be happy to double-check if these are all resolved when you're about to complete this PR.

File: squidasm/sim/stack/netstack.py (hunk 2 above)
Comment (mkvanhooft) [reply squidasm/sim/stack/netstack.py:341]: No, from what I gather, it seems that NetQASM generates the instructions to do the bell state correction. Doing any bell state correction here would give incorrect results for any model with non-default bellIndex

File: pyproject.toml
Comment (mkvanhooft) [inline pyproject.toml:1]: Don't forget to undo this before merge

Comment (mkvanhooft) [inline pyproject.toml:1]: Don't forgot to undo this change

**Review by bvdvecht — APPROVED** [review]
